### ISSUE TITLE
Feedback from first GameJam

### ISSUE THREAD

**Issue by FredTheDino** [lithekod/snake-ribs#25]
**\*Note that this will be split up into more issues, just posting it here for now before I/we forget!\***
 - Camera might be useful. Some people struggled with it.
 - Some way to set the title of the game, would be really nice! :D
 - Try to find a better alternative than pygame?
     - It's hard to install on some systems, if the Python version you use doesn't support the ribs API. (People who use it might have problems with features they expect as well!)
     - Sound doesn't work great, there can be some huge lag!
     -  ~http://pyglet.org/~ 
     - https://arcade.academy/ (<- This is what snake-ribs is ment to be... So, just use this instead?)
 - Change name of the project, to `ribs.py`? Maybe?
 - Make the files less daunting, bigger focus on the what stuff is where! (Some suggested splitting up the files more)
 - Make more people familiar with the code base! :D
 - Fix it so flake8 doesn't report errors in our code...
 - Upload to https://pypi.org
 - A `GAMES.md` (or issue?) with links to created games.

**Comment by TheZoq2** [comment]
> Try to find a better alternative than pygame?

Ah, the age old question. Unfortunately, the answer seems to always be that pygame is probably better.

Though if we're writing a wrapper around things anyway, SDL might be an option.

**Comment by FredTheDino** [comment]
The problem with SDL, is that you have to compile SDL, or a wrapper around it somehow. Or is there a python sdl wrapper allready out there?

While arcade is written entirely in Python, it's one pip install and you're ready to roll. I don't think we can beat that TBH.

**Comment by TheZoq2** [comment]
Yea, arcade looks neat.

I don't remember how SDL worked exactly, it might very well be that you need to do a bunch of woodoo. Perhaps @emiluren remembers more, I think he set things up initially in at least one of my pysdl projects

**Comment by Emiluren** [comment]
Yeah, there is pySDL I think it's called. Since pygame already has a dependency on SDL that shouldn't make a diffference

**Comment by Emiluren** [comment]
One advantage of using SDL directly is you can use SDL 2 with hardware acceleration and don't have to rely on old SDL 1.2 which pygame uses

**Comment by FredTheDino** [comment]
But why would we require SDL if we don't have to? If we can use something completely written in Python, that solves the entire problem, no?

I'm trying to communicate: Let's ***not*** use something based on SDL, since it requires a compiler and other things. Arcade only requires PIP and Python, so it should be easier to get started with. Arcade > SDL

**Comment by FredTheDino** [comment]
https://pysdl2.readthedocs.io/en/rel_0_9_7/integration.html#known-problems-with-sdl2-binaries
As an example

But if you look at something like Arcade:
https://github.com/pythonarcade/arcade/issues/742
They understand that people want to bundle it all together, and they're dedicated to making it work on Mac, Windows and Linux.

As I see it, we have two options.
 1. Write a custom python game library/engine, using PySDL, might be buggy, might have trubble installing on some platforms. It would be cool to have a custom python engine, and we'd have full knowledge of it unless people move, which they might. It's a lot more work to write, and we'd have to solve problems like text rendering in a good way. Totally plausible.
 2. Switch to Arcade, and just maintain a simple example game, maybe contribute to the project aswell. Arcade will be around even if we stop using it, giving the people who use it something they can actually use when not a GameJam. We could put more effort into making good documentation and a good starting point which might benefit other people using Arcade.

I think you know where I stand, lemme know what you think. :D

**Comment by Emiluren** [comment]
Yeah I know nothing about arcade, but I'm not opposed to it

**Comment by FredTheDino** [comment]
Tell me why! Give me a reason!

**Comment by TheZoq2** [comment]
Haven't looked into arcade, but it does sound like a good idea

**Comment by Emiluren** [comment]
> Tell me why! Give me a reason!

A reason for not being opposed?

**Comment by FredTheDino** [comment]
I must have missread, I missed your *not*, which kinda changed the sentence. ;)

**Comment by FredTheDino** [comment]
I think we need a meeting here TBH.

**Comment by TheZoq2** [comment]
> A reason for not being opposed?

Yes. Please write a 1000-2000 word essay about this. It should be on my desk tomorrow at 17
